src/main/java/io/github/genomicdatainfrastructure/daam/exceptions/ApplicationSubmissionException.java

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+package io.github.genomicdatainfrastructure.daam.exceptions;
+
+import java.util.List;
+
+public class ApplicationSubmissionException extends RuntimeException {
+
+    private static final String MESSAGE = "Application %s could not be submitted";
+    private final List<String> errorMessages;
+
+    public ApplicationSubmissionException(Long applicationId, List<String> errorMessages) {
+        super(String.format(MESSAGE, applicationId) + (errorMessages == null || errorMessages
```

Comment:
hint: Now Java support formatting directly from the string object you want a base. It would look like: `MESSAGE.formatted(applicationId)`.